pkg/manager/config/manager.go

```diff
@@ -59,21 +71,61 @@ func (srv *ConfigManager) Init(addrs []string, cfg config.ConfigManager, logger
 
 	etcdClient, err := clientv3.New(etcdConfig)
 	if err != nil {
-		return errors.WithMessage(err, "create etcd config center error")
+		return errors.Wrapf(err, "create etcd config center error")
 	}
 	srv.etcdClient = etcdClient
 	srv.kv = clientv3.NewKV(srv.etcdClient)
 
+	ctx, cancel := context.WithCancel(ctx)
+	srv.cancel = cancel
+
+	srv.initProxyConfig(ctx)
+
 	return nil
 }
 
+func (e *ConfigManager) watch(ctx context.Context, ns, key string, f func(*zap.Logger, *clientv3.Event)) {
+	wkey := path.Join(e.basePath, ns, key)
+	logger := e.logger.With(zap.String("comp", wkey))
+	e.wg.Run(func() {
+		wch := e.etcdClient.Watch(ctx, wkey)
+		ticker := time.Tick(DefaultWatchLease)
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker:
+				resp, err := e.kv.Get(ctx, path.Join(e.basePath, ns, wkey))
```

Comment:
wkey=path.Join(e.basePath, ns, key), you are join e.basePath and ns again?
```suggestion
				resp, err := e.kv.Get(ctx, wkey)
```

Reply:
I think it is only testable by failpoints. To trigger the ticker manually.

Reply:
Yes, it's hard to test. Or you can pass the interval as a parameter and set the interval to 1 second in the test.

Reply:
Added a test for `e.watch()`.